### ISSUE TITLE
Issue with notifications while the iOs application is not running

### ISSUE THREAD

**Issue by sjahan** [phonegap/phonegap-plugin-push#1915]
Hi, 

First of all, thank you very much for your work on this great plugin.
I've got a question about it on **iOs, v2.0.0 (cordova-ios v4.4.0)** because I was not able to determine whether it is the expected behavior or not: 

When the application is down/terminated/killed/not running, the device properly receives the notifications (which is great).
These notifications appears in the notification center/notification shade. 
The problem is that if I start the application **without** tapping the notification, then the application acts as there were no notifications and my 'on notification' handler is not invoked.
If the application is started but in background, this works: I can open the application without tapping the notification, the application receives it anyway.
Is it the expected behavior?

The notification is high/10 priority and the `content-available` equals 1.
What is weird is that when the application boots and there is a notification pending, I get the following in the iPhone logs: 

```
Aug 22 13:43:21 iPhone App[375] <Notice>: createNotificationChecker
Aug 22 13:43:21 iPhone App[375] <Notice>: not coldstart
Aug 22 13:43:21 iPhone App[375] <Notice>: active
Aug 22 13:43:21 iPhone App[375] <Notice>: PushPlugin skip clear badge
Aug 22 13:43:21 iPhone App[375] <Notice>: Launch notification doesn't exist <-- custom logs I added to check if there is a notification or not.
```

I don't have many experiences with native iOs development, but from my understanding, I should see `coldstart` and not `not coldstart`, and I should get a launch notification. These should be provided by the OS, so I assume I must miss something with the payload, but I don't see what.

The phone receives the following notification from its own logs: 

```
Aug 22 14:14:44 iPhone apsd(PersistentConnection)[86] <Notice>: 2017-08-22 14:14:44 +0200 apsd[86]: <APSCourier: 0x166714a0>: Received message for enabled topic 'com.app.mobile.client' with payload '{
    action = "my_action";
    aps =     {
        alert =         {
            body = "Test Body";
            title = "Test Title";
        };
        "content-available" = 1;
    };
    "gcm.message_id" = "0:150340408XXXXXX%67900cecXXXXX";
    profile = "my_profile";
}' onInterface: NonCellular  for device token: NO  with priority 10
```

It currently does work on the Android version of our application, so I don't know whether the Android version has more abilities, or if it could be a bug for the iOs version, or simply some iOs limitations. 
A lot of issues had been opened about this topic, some closed, some still opened and this is very hard to know if it is supposed to work or not currently... so that's why I'm opening this case, I apologize in advance if the information was clearly written down somewhere!

Anyway, thank you for your help on this!

**Comment by sjahan** [comment]
@macdonst did you get some time for my question?

I assume from what I could read that this is actually the normal behavior for iOs, and I'll probably have to fetch the notification 'manually' by calling a webservice dedicated to check/fetch pending notifications, 
but I'd be glad if someone could tell me this behavior is totally expected and that I didn't get it wrong.

**Comment by talvespe** [comment]
@sjahan Hi buddy, I'm facing exactly the same issue.  
ps: On Android it works like a charm! 

Follow below my payload ( I'm also using FCM )
```javascript
{  
   "notification":{  
      "sound":"default",
      "icon":"",
      "body":"My Body Notification ",
      "title":"Tittle Notification"
   },
   "content_available":true,
   "data":{ 
      "content-available":"1",
      "categoriaId":"1",
      "notId":"1"
   },
   "to":"\/topics\/DESENV",
   "priority":"high"
}
```
When I start the application without tapping the notification the application the **on.notification** isn't triggered.

**Log after start without tap the notification**

```javascript
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online(CFNetwork)[2150] <Notice>: Faulting in NSHTTPCookieStorage singleton
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online(CFNetwork)[2150] <Notice>: Faulting in CFHTTPCookieStorage singleton
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online(CFNetwork)[2150] <Notice>: Creating default cookie storage with default identifier
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: Apache Cordova native platform version 4.4.0 is starting.
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: Multi-tasking -> Device: YES, App: YES
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: 0x1051f0000 - DocumentLoader::startLoadingMainResource: Returning empty document (frame = 0x1045ac000, main = 1)
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: Memory usage info dump at MainFrameLoadCompleted:
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: pagecache_page_count: 0
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: compressed: 0
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: javascript_gc_heap_extra_memory_size: 0
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: document_count: 1
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: phys_footprint: 4808288
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: resident_size: 7565312
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: javascript_gc_heap_capacity: 65536
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: virtual_size: 1168900096
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: internal: 4628480
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: Using UIWebView
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: [CDVTimer][handleopenurl] 0.325024ms
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: [CDVTimer][intentandnavigationfilter] 3.316998ms
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: [CDVTimer][gesturehandler] 0.256002ms
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: [CDVTimer][splashscreen] 25.346994ms
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: [CDVTimer][statusbar] 14.288008ms
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: [CDVTimer][keyboard] 1.598001ms
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: [CDVTimer][TotalPluginStartup] 46.965003ms
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: createNotificationChecker
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: not coldstart
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: active
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: PushPlugin skip clear badge
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online(libMobileGestaltExtensions.dylib)[2150] <Notice>: libMobileGestalt MobileGestaltSupport.m:153: pid 2150 (Cidada\M-C\M^C\M-C\M^Io Online) does not have sandbox access for frZQaeyWLUvLjeuEK43hmg and IS NOT appropriately entitled
Sep 12 17:37:18 iPhone-de-Thiago-Pelegrine Cidadão Online(libMobileGestalt.dylib)[2150] <Notice>: libMobileGestalt MobileGestalt.c:550: no access to InverseDeviceID (see <rdar://problem/11744455>)
Sep 12 17:37:19 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: Memory usage info dump at MainFrameLoadStarted:
Sep 12 17:37:19 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: pagecache_page_count: 0
Sep 12 17:37:19 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: compressed: 0
Sep 12 17:37:19 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: javascript_gc_heap_extra_memory_size: 0
Sep 12 17:37:19 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: document_count: 1
Sep 12 17:37:19 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: phys_footprint: 22352392
Sep 12 17:37:19 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: resident_size: 15122432
Sep 12 17:37:19 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: javascript_gc_heap_capacity: 589824
Sep 12 17:37:19 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: virtual_size: 1192214528
Sep 12 17:37:19 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: internal: 10354688
Sep 12 17:37:19 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: 0x1045ac088 - FrameLoader::prepareForLoadStart: Starting frame load (frame = 0x1045ac000, main = 1)
Sep 12 17:37:19 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: Resetting plugins due to page load.
Sep 12 17:37:19 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: 0x1051f1a00 - DocumentLoader::startLoadingMainResource: Starting load (frame = 0x1045ac000, main = 1)
Sep 12 17:37:21 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: Memory usage info dump at MainFrameLoadCompleted:
Sep 12 17:37:21 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: pagecache_page_count: 0
Sep 12 17:37:21 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: compressed: 0
Sep 12 17:37:21 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: javascript_gc_heap_extra_memory_size: 8564849
Sep 12 17:37:21 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: document_count: 1
Sep 12 17:37:21 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: phys_footprint: 157525432
Sep 12 17:37:21 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: resident_size: 150913024
Sep 12 17:37:21 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: javascript_gc_heap_capacity: 59881089
Sep 12 17:37:21 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: virtual_size: 1339703296
Sep 12 17:37:21 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: internal: 145260544
Sep 12 17:37:21 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: 0x1045ac088 - FrameLoader::checkLoadCompleteForThisFrame: Finished frame load (frame = 0x1045ac000, main = 1)
Sep 12 17:37:21 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: Finished load of: file:///var/containers/Bundle/Application/18D0251F-3A6E-48C4-B47D-784DD2C5C85E/Cidada%CC%83o%20Online.app/www/index.html
Sep 12 17:37:21 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: 0x1050fb000 - DocumentLoader::startLoadingMainResource: Returning empty document (frame = 0x10b948480, main = 0)
Sep 12 17:37:21 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: Ionic Native: deviceready event fired after 1855 ms
Sep 12 17:37:21 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: Push Plugin register called
Sep 12 17:37:21 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: PushPlugin.register: setting badge to false
Sep 12 17:37:21 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: PushPlugin.register: clear badge is set to 0
Sep 12 17:37:21 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: PushPlugin.register: better button setup
Sep 12 17:37:21 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: GCM Sender ID 861476504103
Sep 12 17:37:21 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: Using GCM Notification
Sep 12 17:37:21 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: Using GCM Sandbox
Sep 12 17:37:21 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: Push Plugin register success: <45b0e53a 40db1c53 349ddb76 edc76508 724d473a ce8a2847 6dda8af9 57c4404b>
Sep 12 17:37:21 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: GCM Registration Token: ewKP43nyV88:APA91bE9JyaVhtUe9bxT66R0sIXQtZs_AtJRQMuY7RhL3LsLFbWMJK8w0NN8i_N4NZFDmX1_f1cfMh8_vlkAJwZiQQfRyMv37xClbsic7LH4qgwfiMOV64teopczJ8Dhf4cTKKzHSd7x
Sep 12 17:37:22 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: SUBSCRIBE Home
Sep 12 17:37:22 iPhone-de-Thiago-Pelegrine Cidadão Online(AppleJPEG)[2150] <Notice>: [0x1018ca600] Created session
Sep 12 17:37:22 iPhone-de-Thiago-Pelegrine Cidadão Online(AppleJPEG)[2150] <Notice>: [0x1018ca600] Releasing session
Sep 12 17:37:22 iPhone-de-Thiago-Pelegrine Cidadão Online(AppleJPEG)[2150] <Notice>: [0x101078000] Created session
Sep 12 17:37:22 iPhone-de-Thiago-Pelegrine Cidadão Online[2150] <Notice>: runTasks()
Sep 12 17:37:22 iPhone-de-Thiago-Pelegrine Cidadão Online(CFNetwork)[2150] <Notice>: TIC TCP Conn Start [1:0x17419b930]
Sep 12 17:37:22 iPhone-de-Thiago-Pelegrine Cidadão Online(CFNetwork)[2150] <Notice>: TIC TCP Conn Event [1:0x17419b930]: 1 Err(0)
Sep 12 17:37:22 iPhone-de-Thiago-Pelegrine Cidadão Online(CFNetwork)[2150] <Notice>: TIC TCP Conn Connected [1:0x17419b930]: Err(0)
Sep 12 17:37:27 iPhone-de-Thiago-Pelegrine Cidadão Online(AppleJPEG)[2150] <Notice>: [0x101078000] Releasing session
Sep 12 17:37:31 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: 0x100e02150 - PerformanceMonitor::measurePostLoadMemoryUsage: Process was using 129849624 bytes of memory after the page load.
Sep 12 17:37:36 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2150] <Notice>: 0x100e02150 - PerformanceMonitor::measurePostLoadCPUUsage: Process was using 3.6% CPU after the page load.
Sep 12 17:38:48 iPhone-de-Thiago-Pelegrine Cidadão Online(CFNetwork)[2150] <Notice>: TIC TCP Conn Cancel [1:0x17419b930]
Sep 12 17:38:48 iPhone-de-Thiago-Pelegrine Cidadão Online(CFNetwork)[2150] <Notice>: TIC TCP Conn Destroyed [1:0x17419b930]
```

**Log after tapping the notification - just to compare once that this way works fine**
```javascript
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online(CFNetwork)[2162] <Notice>: Faulting in NSHTTPCookieStorage singleton
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online(CFNetwork)[2162] <Notice>: Faulting in CFHTTPCookieStorage singleton
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online(CFNetwork)[2162] <Notice>: Creating default cookie storage with default identifier
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: Apache Cordova native platform version 4.4.0 is starting.
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: Multi-tasking -> Device: YES, App: YES
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: 0x1035f0000 - DocumentLoader::startLoadingMainResource: Returning empty document (frame = 0x1029ac000, main = 1)
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: Memory usage info dump at MainFrameLoadCompleted:
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: pagecache_page_count: 0
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: compressed: 0
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: javascript_gc_heap_extra_memory_size: 0
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: document_count: 1
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: phys_footprint: 4849392
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: resident_size: 7602176
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: javascript_gc_heap_capacity: 65536
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: virtual_size: 1173684224
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: internal: 4661248
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: Using UIWebView
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: [CDVTimer][handleopenurl] 0.209987ms
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: [CDVTimer][intentandnavigationfilter] 3.580987ms
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: [CDVTimer][gesturehandler] 0.092983ms
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: [CDVTimer][splashscreen] 19.990981ms
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: [CDVTimer][statusbar] 11.772037ms
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: [CDVTimer][keyboard] 1.505971ms
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: [CDVTimer][TotalPluginStartup] 38.675010ms
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: createNotificationChecker
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: coldstart
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: didReceiveNotification with fetchCompletionHandler
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: app in-active
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: this should be a silent push
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: Push Plugin notId 1
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: Notification received
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: active
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: PushPlugin skip clear badge
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online(libMobileGestaltExtensions.dylib)[2162] <Notice>: libMobileGestalt MobileGestaltSupport.m:153: pid 2162 (Cidada\M-C\M^C\M-C\M^Io Online) does not have sandbox access for frZQaeyWLUvLjeuEK43hmg and IS NOT appropriately entitled
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online(libMobileGestalt.dylib)[2162] <Notice>: libMobileGestalt MobileGestalt.c:550: no access to InverseDeviceID (see <rdar://problem/11744455>)
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: Memory usage info dump at MainFrameLoadStarted:
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: phys_footprint: 22417928
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: resident_size: 15245312
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: javascript_gc_heap_capacity: 589824
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: virtual_size: 1192804352
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: internal: 10420224
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: 0x1029ac088 - FrameLoader::prepareForLoadStart: Starting frame load (frame = 0x1029ac000, main = 1)
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: Resetting plugins due to page load.
Sep 12 18:09:46 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: 0x1035f1a00 - DocumentLoader::startLoadingMainResource: Starting load (frame = 0x1029ac000, main = 1)
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: Memory usage info dump at MainFrameLoadCompleted:
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: pagecache_page_count: 0
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: compressed: 0
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: javascript_gc_heap_extra_memory_size: 8564910
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: document_count: 1
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: phys_footprint: 163325512
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: resident_size: 153169920
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: javascript_gc_heap_capacity: 39958206
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: virtual_size: 1348386816
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: internal: 147189760
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: 0x1029ac088 - FrameLoader::checkLoadCompleteForThisFrame: Finished frame load (frame = 0x1029ac000, main = 1)
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: Finished load of: file:///var/containers/Bundle/Application/18D0251F-3A6E-48C4-B47D-784DD2C5C85E/Cidada%CC%83o%20Online.app/www/index.html
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online(WebCore)[2162] <Notice>: 0x1034d3d00 - DocumentLoader::startLoadingMainResource: Returning empty document (frame = 0x109f40000, main = 0)
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: Ionic Native: deviceready event fired after 1843 ms
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: Push Plugin register called
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: PushPlugin.register: setting badge to false
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: PushPlugin.register: clear badge is set to 0
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: PushPlugin.register: better button setup
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: GCM Sender ID 861476504103
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: Using GCM Notification
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: Using GCM Sandbox
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: Push Plugin register success: <45b0e53a 40db1c53 349ddb76 edc76508 724d473a ce8a2847 6dda8af9 57c4404b>
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: GCM Registration Token: ewKP43nyV88:APA91bE9JyaVhtUe9bxT66R0sIXQtZs_AtJRQMuY7RhL3LsLFbWMJK8w0NN8i_N4NZFDmX1_f1cfMh8_vlkAJwZiQQfRyMv37xClbsic7LH4qgwfiMOV64teopczJ8Dhf4cTKKzHSd7x
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: SUBSCRIBE Home
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online(AppleJPEG)[2162] <Notice>: [0x11e0c4000] Created session
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online(AppleJPEG)[2162] <Notice>: [0x11e0c4000] Releasing session
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: runTasks()
Sep 12 18:09:49 iPhone-de-Thiago-Pelegrine Cidadão Online(CFNetwork)[2162] <Notice>: TIC TCP Conn Start [1:0x174381ba0]
Sep 12 18:09:50 iPhone-de-Thiago-Pelegrine Cidadão Online(CFNetwork)[2162] <Notice>: TIC TCP Conn Event [1:0x174381ba0]: 1 Err(0)
Sep 12 18:09:50 iPhone-de-Thiago-Pelegrine Cidadão Online(CFNetwork)[2162] <Notice>: TIC TCP Conn Connected [1:0x174381ba0]: Err(0)
Sep 12 18:09:50 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: Notification received
Sep 12 18:09:50 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: Push Plugin key: alert
Sep 12 18:09:50 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: Push Plugin key: sound
Sep 12 18:09:50 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: Push Plugin key: content-available
Sep 12 18:09:50 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: Notification fired: {"additionalData":{"categoriaId":"1","coldstart":true,"title":"Prioridade Normal 1","message":"Not\M-C\M--cia Prioridade normal 1","notId":"1","cgc":"04663904000158","type":"MENSAGEM_PUSH","content-available":"1","gcm.message_id":"0:1505250582205196%a34899d5a34899d5","foreground":false},"message":"Not\M-C\M--cia Prioridade normal 1","title":"Prioridade Normal 1","sound":"default"}
Sep 12 18:09:50 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: Entrou > MESSAGEM_PUSH
Sep 12 18:09:50 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: not foreground
Sep 12 18:09:50 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: not foreground task
Sep 12 18:09:50 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: Push Plugin finish called
Sep 12 18:09:50 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: processing of push data is finished
Sep 12 18:09:50 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: runTasks()
Sep 12 18:09:50 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: task() 1
Sep 12 18:09:50 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: Push Plugin stopBackgroundTask called
Sep 12 18:09:50 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: Push Plugin handlerObj
Sep 12 18:09:50 iPhone-de-Thiago-Pelegrine Cidadão Online[2162] <Notice>: WARN: (getRootNav) is deprecated and will be removed in the next major release. Use getRootNavById instead.
```

**Comment by sjahan** [comment]
Hi @talvespe, 

I didn't find again where I read it, but as I mentionned in my last message, I think this is a normal iOs behavior, therefore, it can't and won't be fixed (I didn't get any confirmation though).

I encountered another problem on some iPhones, the plugin was unable to register to FCM for some libMobileGestalt reason... As everyone stated on stackoverflow, I switched to cordova-plugin-fcm. With this plugin, you never get notification without tapping them, which makes the situation simpler. 

I'll fetch the pending notifications on the "on resume" event of the application so that it is exactly the same behavior on every platform. 

Hoping this helps a bit!

As I switched and do not use this plugin anymore, I'm going to close this issue.
